modules/core/src/core/model.js

```diff
@@ -80,11 +83,18 @@ export default class Model extends BaseModel {
     return this;
   }
 
-  // TODO - just set attributes, don't hold on to geometry
   setGeometry(geometry) {
     this.drawMode = geometry.drawMode;
     this.vertexCount = geometry.getVertexCount();
-    this.vertexArray.setAttributes(getBuffersFromGeometry(this.gl, geometry));
+
+    for (const name in this.geometryBuffers) {
```

Comment:
Suggest putting the delete code in a helper method with a self documenting name `_deleteGeometryBuffers()`?